honeybee/boundarycondition.py

```diff
@@ -1,116 +1,174 @@
-
-    @boundary_condition_object.setter
-    def boundary_condition_object(self, bco):
-        self._boundary_condition_object = bco
-
     @property
     def view_factor(self):
+        """Get or set the view factor to the ground as a number or 'autocalculate'."""
         return self._view_factor
 
     @view_factor.setter
     def view_factor(self, vf):
         if vf == 'autocalculate':
             self._view_factor = vf
         else:
-            self._view_factor = float(vf)
+            self._view_factor = float_in_range(vf, 0.0, 1.0, 'view factor to ground')
 
     @property
     def sun_exposure_idf(self):
+        """Text string for sun exposure, which is write-able into an IDF."""
         return 'NoSun' if not self.wind_exposure else 'SunExposed'
 
     @property
     def wind_exposure_idf(self):
+        """Text string for wind exposure, which is write-able into an IDF."""
         return 'NoWind' if not self.wind_exposure else 'WindExposed'
 
-    @property
-    def boundary_condition_object_idf(self):
-        return '' if not self.boundary_condition_object else \
-            self.boundary_condition_object.name
-
     def to_dict(self, full=False):
         """Boundary condition as a dictionary.
-        
-        args:
-            full: Set to True to get the full dictionary which includes energy simulation
-            specific keys such as sun_exposure, wind_exposure and view_factor. Default is
-            set to False.
-        
-        returns:
-            A dictionary.
+
+        Args:
+            full: Set to True to get the full dictionary which includes energy
+                simulation specific keys such as sun_exposure, wind_exposure and
+                view_factor. Default: False.
         """
         if full:
-            return {
-                'name': self.name,
-                'bc_object': self.boundary_condition_object_idf,
+            bc_dict = {
+                'type': self.name,
                 'sun_exposure': self.sun_exposure_idf,
                 'wind_exposure': self.wind_exposure_idf,
                 'view_factor': self.view_factor
             }
         else:
-            return {
-                'name': self.name,
-                'bc_object': self.boundary_condition_object_idf
+            bc_dict = {
+                'type': self.name,
             }
+        try:
+            bc_dict['bc_object'] = self.boundary_condition_object
+        except AttributeError:  # Not a Surface boundary condition
+            pass
+        return bc_dict
+
+    def duplicate(self):
+        """Get a copy of this boundary condition."""
+        return self.__copy__()
+
+    def ToString(self):
+        return self.__repr__()
+
+    def __copy__(self):
+        new_bc = self.__class__()
+        new_bc._sun_exposure = self._sun_exposure
+        new_bc._wind_exposure = self._wind_exposure
+        new_bc._view_factor = self._view_factor
+        return new_bc
+
+    def __eq__(self, other):
```

Comment:
Don't you need to check other properties for equality?

Reply:
At first, I thought I might but now I see that it really is not necessary.  I'll remove it.